qiskit/circuit/add_control.py

```diff
@@ -250,20 +250,40 @@ def apply_basic_controlled_gate(circuit, gate, controls, target):
             elif theta == 0 and phi == 0:
                 circuit.mcp(lamb, controls, target)
             else:
-                circuit.mcp(lamb, controls, target)
+                circuit.mcrz(lamb, controls, target, use_basis_gates=True)
                 circuit.mcry(
                     theta,
                     controls,
                     target,
                     use_basis_gates=True,
                 )
-                circuit.mcp(phi, controls, target)
+                circuit.mcrz(phi, controls, target, use_basis_gates=True)
+                circuit.mcp((phi + lamb) / 2, controls[1:], controls[0])
 
     elif gate.name == "z":
         circuit.h(target)
         circuit.mcx(controls, target)
         circuit.h(target)
 
+    elif gate.name == "y":
+        circuit.sdg(target)
+        circuit.mcx(controls, target)
+        circuit.s(target)
+
+    elif gate.name == "h":
+        circuit.mcry(pi / 2, controls, target, use_basis_gates=True)
+        circuit.mcx(controls, target)
+
+    elif gate.name == "sx":
+        circuit.h(target)
+        circuit.mcp(pi / 2, controls, target)
+        circuit.h(target)
+
+    elif gate.name == "sxdg":
+        circuit.h(target)
+        circuit.mcp(3 * pi / 2, controls, target)
+        circuit.h(target)
+
```

Comment:
Are these 4 new ``elif`` statements covered by existing tests (you can check the coverage report for that)?